.github/actions/compute-sha256/action.yml

```diff
@@ -15,17 +15,15 @@ runs:
     - name: Compute the sha256
       id: compute
       shell: bash
-      env:
-        UNTRUSTED_PATH: "${{ inputs.path }}"
       run: |
         set -euo pipefail
 
-        echo "Computing SHA256 for $UNTRUSTED_PATH"
-        if ! [[ -f "$UNTRUSTED_PATH" ]]; then
-            echo "File $UNTRUSTED_PATH not present"
+        echo "Computing SHA256 for $BUILDER_RELEASE_BINARY"
+        if ! [[ -f "$BUILDER_RELEASE_BINARY" ]]; then
+            echo "File $BUILDER_RELEASE_BINARY not present"
             exit 5
         fi
-        digest=$(sha256sum "$UNTRUSTED_PATH" | awk '{print $1}')
+        digest=$(sha256sum "$BUILDER_RELEASE_BINARY" | awk '{print $1}')
```

Comment:
Where is `BUILDER_RELEASE_BINARY` set? I don't believe this will work. Also, we want to use the `path` input for the action as this is intended to be a generic reusable action.

Reply:
I'm not sure I follow.

This is a completely separate GitHub action (`compute-sha256`) so I'm pretty sure that the env vars won't carry over from the `generate-builder` action. Those env vars are only available to the "Generate builder" step in the `generate-builder` action.

I would need more info on what you are doing to be able to tell why you got that error. Could you submit an issue with more details including a link to the workflow you are trying to run?

Reply:
I think I understand a bit better now. The `BUILDER_BINARY` is set by the [container workflow](https://github.com/slsa-framework/slsa-github-generator/blob/1a55da84338004ca26d38b941483c89ab93a695d/.github/workflows/generator_container_slsa3.yml#L19) having a different name than the other workflows.

`BUILDER_BINARY` is not used by the `generate-builder` action. It's simply [set as an input to the action](https://github.com/slsa-framework/slsa-github-generator/blob/1a55da84338004ca26d38b941483c89ab93a695d/.github/workflows/generator_container_slsa3.yml#L99) and should be fine. `BUILDER_RELEASE_BINARY` is set to that input [here](https://github.com/slsa-framework/slsa-github-generator/blob/1a55da84338004ca26d38b941483c89ab93a695d/.github/actions/generate-builder/action.yml#L48). So I think the error you are getting is a result of deleting the env vars from the `generate-builder` action below.

Reply:
BUILDER_BINARY is set as the input but doesn't [this line in the generate-builder action](https://github.com/slsa-framework/slsa-github-generator/blob/1a55da84338004ca26d38b941483c89ab93a695d/.github/actions/generate-builder/action.yml#L48) include it which is then used as an [input for compute-sha256](https://github.com/slsa-framework/slsa-github-generator/blob/1a55da84338004ca26d38b941483c89ab93a695d/.github/actions/generate-builder/action.yml#L68). But on the other side generate-builder.sh is building the executable with the name $BUILDER_RELEASE_BINARY. 

Flow:
generator_container_slsa3.yml (BUILDER_BINARY: generic-generator, which is used as an input for generate builder binary: "${{ env.BUILDER_BINARY }}")
--> generate-builder/action.yml (path: "${{ inputs.binary }}" uses it as an input for compute-sha256)
--> compute-sha256/action.yml (gets path as input UNTRUSTED_PATH: "${{ inputs.path }}" and uses it to look for the executable if ! [[ -f "$UNTRUSTED_PATH" ]]; then...)

We assume that the error 
```
Computing SHA256 for generic-generator
File generic-generator not present
```
results from the executable being generated with the name slsa-generator-generic-linux-amd64 and compute-sha256 is [looking for a file with the name generic-generator](https://github.com/slsa-framework/slsa-github-generator/blob/1a55da84338004ca26d38b941483c89ab93a695d/.github/actions/compute-sha256/action.yml#L24).

Reply:
Normally the way you describe it is true.

1. [`BUILDER_BINARY` is set to `generic-generator`](https://github.com/slsa-framework/slsa-github-generator/blob/1a55da84338004ca26d38b941483c89ab93a695d/.github/workflows/generator_container_slsa3.yml#L19)
2. [`BUILDER_BINARY` is used as input into `generate-builder`](https://github.com/slsa-framework/slsa-github-generator/blob/1a55da84338004ca26d38b941483c89ab93a695d/.github/workflows/generator_container_slsa3.yml#L99)
3. [`generate-builder` sets the input value to `BUILDER_RELEASE_BINARY`](https://github.com/slsa-framework/slsa-github-generator/blob/1a55da84338004ca26d38b941483c89ab93a695d/.github/actions/generate-builder/action.yml#L48)
4. [`generate-builder.sh` uses `BUILDER_RELEASE_BINARY` as the name of the binary to build. ](https://github.com/slsa-framework/slsa-github-generator/blob/1a55da84338004ca26d38b941483c89ab93a695d/.github/actions/generate-builder/generate-builder.sh#L13)

However in your PR you [removed](https://github.com/slsa-framework/slsa-github-generator/pull/606/files#diff-bff0b44f6cbd4c751293462017ea9ce8c048288fad10886047a63503ff7112adL46) the environment variables set in `generate-builder`. This causes the value of `BUILDER_RELEASE_BINARY` to be the value `slsa-generator-generic-linux-amd64` inherited from the workflow as set [here](https://github.com/slsa-framework/slsa-github-generator/blob/1a55da84338004ca26d38b941483c89ab93a695d/.github/workflows/generator_container_slsa3.yml#L20). So the `generate-builder` action builds the builder with the name `slsa-generator-generic-linux-amd64`.

Then the [next step](https://github.com/slsa-framework/slsa-github-generator/blob/1a55da84338004ca26d38b941483c89ab93a695d/.github/actions/generate-builder/action.yml#L64-L68) that generates the sha256 hash of the binary accepts `inputs.binary` as the binary name which is `generic-generator`. This is where your error is occurring. The `compute-sha256` action is expecting to find a binary named `generic-generator` but the name of the binary that was created is  `slsa-generator-generic-linux-amd64`.

If you [restore the environment variables](https://github.com/slsa-framework/slsa-github-generator/pull/606/files#diff-bff0b44f6cbd4c751293462017ea9ce8c048288fad10886047a63503ff7112adL46) in the `generate-builder` action. This error should go away.

Reply:
You are right, we just verified it with the other adaptions we made. Thank you for taking the time to look into this! 
If you're interested, [this run](https://github.com/Root-DE/Example-Image/actions/runs/2718167924) finished successfully, pushed image and signed attestation. We also used the cosign verifier to verify the signed attestation which works.